File: src/pt/up/fe/luisfonseca/cp/api/StationsLoader.java
package pt.up.fe.luisfonseca.cp.api;

import pt.up.fe.luisfonseca.cp.api.json.Stations;

import com.google.gson.Gson;

public class StationsLoader extends RetrieveTask<String[]> {
	
	private static class StationsParser implements JsonParser<String[]>
	{
		public String[] parse(String json) {
	        Gson gson = new Gson();
	        Stations response = gson.fromJson(json, Stations.class);
	        
			return response.stations;
		}
	}

	public StationsLoader(ResponseHandler<String[]> handler) {
		super(handler, new StationsParser());
		this.execute(CP.URLstations);
	}
}

File: src/pt/up/fe/luisfonseca/cp/api/StationsLoader.scala
package pt.up.fe.luisfonseca.cp.api

import scala.util.parsing.json.JSON
import pt.up.fe.luisfonseca.cp.api.json.Stations
import org.json.JSONObject

class StationsParser extends JsonParser[List[String]]
{
	def parse(json: String) : List[String] = {
	  
	  val jObject = new JSONObject(json);
	  if (jObject.has("stations"))
	  {
		  val jStations = jObject.getJSONArray("stations")
		  
		  for(i<- (0 to jStations.length() - 1).toList)
		    yield jStations.getString(i)
	  }
	  else
	    null
	}
}

class StationsLoader(val handler: ResponseHandler[List[String]])
	extends RetrieveTask[List[String]](handler, new StationsParser())
{
	def execute = super.execute(CP.URLstations);
}